hash-table/src/hash_table.c

```diff
@@ -180,8 +177,8 @@ char* ht_search(ht_hash_table* ht, char* key) {
  * Deletes key's item from the hash table. Does nothing if 'key' doesn't exist
  */
 void ht_delete(ht_hash_table* ht, char* key) {
-    float load = (float)ht->count / ht->size;
-    if (load < 0.1) {
+    int load = ht->count*100 / ht->size;
```

Comment:
Avoid `float` in simple C, if easy/reasonable/possible.  (Many embedded platforms only have soft-floating-point).